pkg/network/connection.go

```diff
@@ -850,13 +851,29 @@ func (c *connection) writeBufLen() (bufLen int) {
 	return
 }
 
+// Close Connection after 1 second (to avoid RST) after sending GoAway to client.
+// 1 second is borrow from go.
+var delayTime = 1 * time.Second
+
 func (c *connection) Close(ccType api.ConnectionCloseType, eventType api.ConnectionEvent) error {
 	defer func() {
 		if p := recover(); p != nil {
 			log.DefaultLogger.Errorf("[network] [close connection] panic %v\n%s", p, string(debug.Stack()))
 		}
 	}()
 
+	// always stop the previous timer.
+	if t := c.shutdownTimer; t != nil {
```

Comment:
I think it's safer to always stop the previous timer, since there may have another `Close(NoFlush)` after `Close(DelayClose)`.
Also, it's safe to call `Stop` on an expired timer.